pkg/config/server.go

```diff
@@ -29,29 +30,56 @@ type UserRolesConfig struct {
 type UserConfig struct {
 	PublicKey string   `mapstructure:"public_key"`
 	Name      string   `mapstructure:"name"`
+	GithubID  string   `mapstructure:"github_id"`
 	Targets   []string `mapstructure:"targets"`
 }
 
 func (uc *UserConfig) ID() string {
 	return base64.StdEncoding.EncodeToString([]byte(uc.PublicKey))
 }
 
+type GithubConfig struct {
+	APIEndpoint string `mapstructure:"api_endpoint"`
+}
+
 type ServerConfig struct {
 	Port    int                         `yaml:"port"`
 	Targets map[string]*UserRolesConfig `mapstructure:"targets"`
 
-	Users []*UserConfig `mapstructure:"users"`
+	Github *GithubConfig `mapstructure:"github"`
+	Users  []*UserConfig `mapstructure:"users"`
 }
 
 func (sc *ServerConfig) NewAuthorizer() (*auth.Authorizer, error) {
 	users := map[string]*rsa.PublicKey{}
 	for _, user := range sc.Users {
 		pub, err := auth.DecodeAuthorizedKey([]byte(user.PublicKey))
 		if err != nil {
-			return nil, fmt.Errorf("Failed to decode user public key: %v, %#v", err, user.PublicKey)
+			fmt.Printf("[WARN] Failed to decode user public key: %v, %#v\n", err, user.PublicKey)
 		}
 		users[user.ID()] = pub
+		if user.GithubID != "" {
```

Comment:
I think the implemented logic loops over all users once, requesting github keys for those who have it configured. 
 Instead, [I was suggesting](https://github.com/superorbital/cludo/issues/47#issuecomment-1015956479) that we should loop over users twice - once to check if there's any matching key configured, and then (if there isn't) again to ask github for each configured key.  That way we never ping github if there's a local match.

Reply:
done